iothub/device/src/Transport/Amqp/AmqpConnectionPool.cs

```diff
@@ -154,5 +148,12 @@ private AmqpConnectionHolder ResolveConnectionByHashing(AmqpConnectionHolder[] p
 
             return pool[index];
         }
+
+        private static int GetDeviceIdentityIndex(DeviceIdentity deviceIdentity, int poolLength)
+        {
+            return deviceIdentity == null
```

Comment:
This is a good null check to have, but by design we would not be getting the index of a device identity that is null, right?

Reply:
that is correct, that's why it will throw an exception indicating something went wrong